### PR TITLE
Factorize `setClosingOnClick` listeners

### PR DESCRIPTION
**Pull request by FabianDevel**
Create a parent class for all actions bottomsheet of the app

Depends on #637

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt [hunk 1]
```diff
@@ -17,134 +17,24 @@
...
 
-    private fun MainActionsView.setClosingOnClickListener(callback: ((Int) -> Unit)) {
+    protected fun MainActionsView.setClosingOnClickListener(callback: (Int) -> Unit) {
         setOnItemClickListener { id ->
-            callback(id)
             findNavController().popBackStack()
+            callback(id)
         }
     }
 }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:40]: Are we sure that's the correct order for all situations ?

By order, I mean `callback()` then `popBackStack()`, or `popBackStack()` then `callback()`.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt [hunk 2]
```diff
@@ -17,134 +17,24 @@
...
             callback()
         }
     }
 
-    private fun MainActionsView.setClosingOnClickListener(callback: ((Int) -> Unit)) {
+    protected fun MainActionsView.setClosingOnClickListener(callback: (Int) -> Unit) {
         setOnItemClickListener { id ->
-            callback(id)
             findNavController().popBackStack()
+            callback(id)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:37]: Are we sure that's the correct order for all situations, for both `setClosingOnClickListener()` ?

By order, I mean `callback()` then `popBackStack()`, or `popBackStack()` then `callback()`.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*
  * Infomaniak kMail - Android
- * Copyright (C) 2022-2023 Infomaniak Network SA
+ * Copyright (C) 2023 Infomaniak Network SA
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:3]: Do we want this ?

Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:3]: Yes it's technically a new file, the old one has been renamed

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:37]: It's popBackStack() then callback for both

**Review by KevinBoulongne — APPROVED** [review]

**Review by LunarX — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt [hunk 4]
```diff
@@ -59,21 +62,15 @@ class DetailedContactBottomSheetDialog : BottomSheetDialogFragment() {
     }
 
     private fun setupListeners() = with(binding) {
-        writeMail.setOnClickListener {
+        writeMail.setClosingOnClickListener {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt:65]:
```suggestion

        writeMail.setClosingOnClickListener {
```

**Pull request merged**